### PR TITLE
[#9534] Deleting the last instructor in the course leaves a dangling course object in the database

### LINKED ISSUE #9534: Deleting the last instructor in the course leaves a dangling course object in the database.
- **Environment**: master branch at commit 2429ab644d751ff8e2fc310cd6dc854f40992868

**Steps to reproduce**
* Login as instructor _A_
* create a course _B_.
* Login as admin
* Visit http://localhost:4200/web/admin/accounts?instructorid=A (For example, http://localhost:4200/web/admin/accounts?instructorid=test@example.com)
* Choose to delete instructor _A_ from course _B_

**Actual behavior**
* Deletion success. Course _B_ now has no instructor associated to it, so it basically becomes a dangling course object in the database with no way to access it.

**Expected behaviour**
Perhaps change it so that deleting the last instructor from a course isn't possible? This would allow utilizing the same functionality for an instructor - if an instructor tries to delete himself from a course in which he's the only instructor, that fails with the message `The instructor you are trying to delete is the last instructor in the course. Deleting the last instructor from the course is not allowed.` 

**Additional info**
Looking for feedback on if you want this feature or not @damithc

### PR DESCRIPTION
**Pull request by mattlim1207**
Fixes #9534

**Outline of Solution**

1. Removed the truth check in `DeleteInstructorAction` that allowed admins to bypass the truth checking stopping the last instructor of a course from being deleted from said course.
2. Modified the corresponding unit test, `testExecute_deleteLastInstructorByGoogleId_shouldPass`, by flipping asserts, and renaming accordingly.

### CODE REVIEW

**Comment by mattlim1207** [comment]
Just noticed end-to-end testing needs to be updated as well, working on that now

**Comment by mattlim1207** [comment]
The only E2E test that should be affected now passes, and the consistently failing test seems to fail both on my feature branch and master

**Comment by NicolasCwy** [comment]
@mattlim1207 will be checking this out

File: src/e2e/resources/data/AdminAccountsPageE2ETest.json [hunk 1]
```diff
@@ -96,6 +102,29 @@
...
         "sessionLevel": {}
       }
+    },
+    "AAccounts.instr3-AAccounts.CS2103": {
+      "googleId": "tm.e2e.AAccounts.instr3",
+      "courseId": "tm.e2e.AAccounts.CS2103",
+      "name": "Teammates Instr3",
+      "email": "AAccounts.instr3@gmail.tmt",
+      "role": "Co-owner",
+      "isDisplayedToStudents": true,
```
[the remark this answers is not in the document]
Comment (mattlim1207) [reply src/e2e/resources/data/AdminAccountsPageE2ETest.json:112]: So from what I gathered about E2E tests and the "happy path," I figured a test should check the admin can successfully delete an instructor, as deleting a last instructor of a course is more of an edge case. Thus, it required another instructor to be created

File: src/e2e/resources/data/AdminAccountsPageE2ETest.json [hunk 2]
```diff
@@ -57,7 +63,7 @@
       "name": "Teammates Instr2",
       "email": "AAccounts.instr2@gmail.tmt",
       "role": "Co-owner",
-      "isDisplayedToStudents": false,
+      "isDisplayedToStudents": true,
```
[the remark this answers is not in the document]
Comment (mattlim1207) [reply src/e2e/resources/data/AdminAccountsPageE2ETest.json:66]: As mentioned above, I wanted to test that the admin could in fact delete an instructor. Because the built-in requirement is that a visible instructor is left, I changed the original default instructor to be visible, so the other one would be deleted successfully.

Also, as far as I understand, it makes more sense for the instructor to be visible, as an admin can't even delete instructors if there aren't any visible ones left (making it seem to me that it's assumed there needs to be one visible instructor of a course)

File: src/e2e/resources/data/AdminAccountsPageE2ETest.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (mattlim1207) [reply src/e2e/resources/data/AdminAccountsPageE2ETest.json:112]: It's the AdminAccountsPageE2ETest. I figured I should fix the issue, and all associated unit tests and E2E tests, in one PR. The E2E test was set up to create a course with one invisible instructor, then simulate the admin deleting the instructor. Because this behavior was changed with this PR, I modified the initial parameters.

[the remark this answers is not in the document]
Comment (NicolasCwy) [reply src/e2e/resources/data/AdminAccountsPageE2ETest.json:112]: I see, so the changes you have made would cause AdminAccountsPageE2ETest to fail as the there is currently 1 instructor attached to the course CS2103 (they are invisible as well, causing the test to fail too).

Thanks for clarifying, this makes sense to me

**Review by NicolasCwy — APPROVED** [review]
LGTM! Nice work and thanks for contributing

**Review by ziqing26 — APPROVED** [review]
LGTM!
